hoomd/mpcd/DryingDropletStreamingMethod.cc

```diff
@@ -0,0 +1,132 @@
+                                                                 int phase,
+                                                                 std::shared_ptr<::Variant> R,
+                                                                 mpcd::detail::boundary bc,
+                                                                 Scalar density,
+                                                                 unsigned int seed)
+    : mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::detail::SphereGeometry>()),
+      m_R(R), m_bc(bc), m_density(density), m_seed(seed), m_picks(m_exec_conf), m_picker(m_sysdef, seed)
+    {}
+
+/*!
+ * \param timestep Current time to stream
+ */
+void mpcd::DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    // use peekStream since shouldStream will be called by parent class
+    if(!peekStream(timestep)) return;
+
+    // compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_mpcd_dt);
+    // make sure that V < 0, since size of droplet must decrease
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (!m_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc );
+        validate();
+        m_validate_geom = false;
+        }
+
+    /*
+     * Update the geometry radius to the size at the end of the streaming step.
+     * This needs to be done every time.
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+
+    // stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    // calculating number of particles to evaporate(such that solvent density remain constant)
+    const unsigned int N_end = std::round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const unsigned int N_global = m_mpcd_pdata->getNGlobal();
+    const unsigned int N_evap = (N_end < N_global) ? N_global - N_end : 0;
+
+    /*
+     * Picking N_evap particles out of total number of bounced particles using RandomPicker,
+     * m_Npick is the total number of particles picked on this rank, m_picks will contain the indices of
+     * picked particles in \a m_bounced array.
+     */
+    m_Npick = 0;
+    m_picker(m_picks, m_Npick, m_bounced, N_evap, timestep, m_mpcd_pdata->getNGlobal());
+
+    /*
+     * applying the picks, In m_bounced array, the particles which were picked are marked 
+     * by setting an additional bit.
+     */
+    applyPicks();
```

Comment:
This snippet of code is so small now, I would just move it here and take away the function call. The code is easier to read that way. Basically just copy lines 108-121 here (including the curly braces).

Once you do that, I think `m_Npick` doesn't need to be a member variable of the class anymore. I would switch that to local scope, at line 83: `unsigned int Npick = 0;`.